tempesta_fw/sock_srv.c

```diff
@@ -434,33 +430,76 @@ static const SsHooks tfw_sock_srv_ss_hooks = {
  * If a server connection is closed at the time this function runs, then
  * it had been closed by a backend before the shutdown, and the connection
  * is still in failover (not re-connected yet). The resources attached to
- * the connection had not been released, and it has to be done forcefully.
+ * the connection may had not been released, and it has to be done forcefully.
  */
 static int
 tfw_sock_srv_disconnect(TfwConn *conn)
```

Comment:
The function always returns `0`, so it can be `void`. Frankly, the non-success return code from `disconnect()` is quite a strange thing.

Reply:
There may be one case when the function can return non-zero value: if `tfw_connection_close()` failed, so we cannot stop connection in this case, and should return error code.